src/splitInteger.js

```diff
@@ -1,21 +1,42 @@
+'use strict';
+
 /**
- * @param {number} value
- * @param {number} numberOfParts
+ * Splits a positive integer into an array of integer parts.
  *
- * @returns {number[]}
+ * @param {number} value - The integer value to split.
+ * @param {number} numberOfParts - The number of parts to split the value into.
+ * @returns {Array<number>} - An array containing exactly number
  */
 function splitInteger(value, numberOfParts) {
-  const parts = [];
-  let rest = value;
+  // Initialize an array with zeros
+  const parts = new Array(numberOfParts).fill(0);
+
+  // If value is less than numberOfParts, fill the array with incre
+  if (value < numberOfParts) {
+    for (let i = 0; i < value; i++) {
+      parts[i] = i + 1; // Fill with values starting from 1
+    }
+
+    return parts;
+  }
+
+  // Calculate the base value for each part
+  const baseValue = Math.floor(value / numberOfParts);
+  const remainder = value % numberOfParts;
 
-  for (let partsLeft = numberOfParts; partsLeft > 0; partsLeft--) {
-    const part = Math.floor(rest / partsLeft);
+  // Fill the array with base values
+  for (let i = 0; i < numberOfParts; i++) {
+    parts[i] = baseValue;
+  }
 
-    parts.push(part);
-    rest -= part;
+  // Distribute the remainder
+  for (let i = 0; i < remainder; i++) {
+    parts[i]++; // Increment the first 'remainder' elements
   }
 
-  return parts;
+  // Sort the parts in ascending order
+  return parts.sort((a, b) => a - b);
```

Comment:
The sorting step here is unnecessary because the distribution of the remainder already ensures that the parts are in ascending order. Removing the sort operation will improve performance.